ingestion-beam/src/main/java/com/mozilla/telemetry/decoder/ParsePayload.java

```diff
@@ -89,15 +89,15 @@ protected PubsubMessage processElement(PubsubMessage message)
       throw e;
     }
 
+    // In case this message is being replayed from an error output where AddMetadata has already
+    // been applied, we strip out any existing metadata fields and put them into attributes.
+    AddMetadata.stripPayloadMetadataToAttributes(attributes, json);
+
     if (MessageScrubber.shouldScrub(attributes, json)) {
       // Prevent the message from going to success or error output.
       throw new MessageShouldBeDroppedException();
     }
 
-    // In case this message is being replayed from an error output where AddMetadata has already
-    // been applied, we strip out any existing metadata fields and put them into attributes.
-    AddMetadata.stripPayloadMetadataToAttributes(attributes, json);
-
```

Comment:
This is a semi-related change as I'm scrutinizing order of operations. We may want to be able to scrub messages when reading from previously decoded output, in which case we probably want attributes to be available.